### DISCUSSION TITLE
Camera Segmentation view with unique colors?

### DISCUSSION THREAD

**Discussion by CaptnMorgan833** [carla-simulator/carla#4153] [Q&A]
Hello,

is it possible to get a camera segementation view in Carla, which displays a unique color for each vehicle (color linked to a specific vehicle ID)?

**Comment by germanros1987** [comment]
The semantic segmentation shader doesn't provide instance segmentation capabilities. You could do that by mixing that mode with the information provided by other modules, such as the BoundingBox module, which provides instance-level information of every object in the world.

E.g.

```# Create the arrays to store the bounding boxes
all_bbs = []
filtered_bbs = []

# Return the bounding boxes of all the elements in the scene
all_bbs = world.get_level_bbs()

# Return the bounding boxes of the buildings in the scene
filtered_bbs = world.get_level_bbs(carla.CityObjectLabel.Buildings)
```

Hope this idea helps.

You can also create an instance segmentation sensor. Contributions are welcome!

**Reply by CaptnMorgan833** [reply]
Thanks for your reply! I will try that.
